tests/checkbox/read-checkbox-grouping.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Group role, name and boundaries are announced when navigating through checkbox group.</title>
+<link rel="author" title="Valerie Young" href="mailto:valerie@bocoup.com">
+<link rel="help" href="https://www.w3.org/TR/wai-aria-practices-1.1/#checkbox">
+<link rel="help" href="https://w3c.github.io/aria/#checkbox">
+<link rel="help" href="https://w3c.github.io/aria/#aria-checked">
+<link rel="help" href="https://w3c.github.io/aria/#group">
+<script src="/resources/ariaatharness.js">
+<script>
+
+  openTestPage("reference/checkbox-group.html");
+
+  presentATTest({
+    mode: ["reading", "interaction"],
+    action: "navigation",
+    assertion: [
+      "The role 'group' is announced",
+      "The group's name 'Sandwich Condiments' is announced",
+      "The boundaries of the group (before the first checkbox and after the last checkbox) are communicated"
+    ],
+    onPartialFailure: (failedInstruction) => { return `
+       <div>
+         <input type="checkbox" id="role-${failedInstruction}" name="role-${failedInstruction}" value="Role was not announced.">
+         <label for="role-${failedInstruction}">Role was not announced.</label>
+       </div>
+       <div>
+         <input type="checkbox" id="name-${failedInstruction}" name="name-${failedInstruction}" value="Name was not announced.">
+         <label for="name-${failedInstruction}">Name was not announced.</label>
+       </div>
+       <div>
+         <input type="checkbox" id="boundaries-${failedInstruction}" name="state-${boundariesInstruction}" value="Boundaries were not announced.">
+         <label for="state-${failedInstruction}">Boundaries were not communicated.</label>
+       </div>
```

Comment:
Consider tracking what the output was that resulted in a failure.

For this, you might need:
 1. the exact command used
 2. the exact output (or summary of output)
 3. Other relevant information. For example, did this happen when reading forwards through the document or backwards? For boundaries, what part of the boundary was not conveyed (enter or exit), etc.

Reply:
So in this iteration, I imagine that presentATTest initially provides "passed" or "failed" below the instructions of the test. If the tester clicks 'failed', then the presentATTest provides the list of commands (perhaps as checkboxes) for the tester to select the exact command that failed. That commands is the "failedInstruction" that is passed as an argument to this function. Selecting a command as failing would then reveals these extra questions. I think I should rename the "failedInstruction" as "failedCommand", but it does assume just one command (rather than a series of commands, I don't know if our tests will have series of commands). Does this cover your concern for exact command used?

That is a really good point about boundaries, I'll just added it to this test. As for exact output I wonder if that is something we would want for every failed command, so we don't need to supply it in this function (because it can be hard coded into the logic of presentATTest).

Reply:
So in this iteration, I imagine that presentATTest initially provides "passed" or "failed" below the instructions of the test. If the tester clicks 'failed', then the presentATTest provides the list of commands (perhaps as checkboxes) for the tester to select the exact command that failed. That commands is the "failedInstruction" that is passed as an argument to this function. Selecting a command as failing would then reveals these extra questions. I think I should rename the "failedInstruction" as "failedCommand", but it does assume just one command (rather than a series of commands, I don't know if our tests will have series of commands). Does this cover your concern for exact command used?

That is a really good point about boundaries, I'll just added it to this test. As for exact output I wonder if that is something we would want for every failed command, so we don't need to supply it in this function (because it can be hard coded into the logic of presentATTest).

Reply:
Yes, I think that makes sense. One question that I have... if multiple commands fail, but each yield different output, is it beneficial to record the output for each failed command?

Reply:
There might be no way of knowing until we've written enough tests and run though them -- but this is a variable we should remember to play around with! Seems like each command could fail differently, though.